be/src/controllers/rest.router.ts

```diff
@@ -65,13 +66,20 @@ router.put('/vocab/:id', (req: Request, res: Response, next: NextFunction) => {
 
 router.get('/vocab/:id', (req: Request, res: Response, next: NextFunction) => {
   vocabularyService.getVocabular(req.params.id)
-    .then(v => vocabDbo2Dto(v))
-    .then(v => res.json(v))
-    .catch(next);
+      .then(v => vocabDbo2Dto(v))
+      .then(v => res.json(v))
+      .catch(next);
 });
 
-router.delete('/vocab/:id', (req: Request, res: Response, next: NextFunction) => {
-  next(new KnowledgeError(501, 'Not Implemented', 'DELETE /vocab/:id is not implemented'));
+router.delete('/vocab/:id', async (req: Request, res: Response, next: NextFunction) => {
+  const header = req.header('if-unmodified-since')
+  const date: Date = new Date(header)
+
+  vocabularyService.deleteVocab(req.params.id, date).then(result => {
+    if (result) {
+      res.status(204).end();
+    }
```

Comment:
What happens when then result is falsy? If it can happen an `else` is required, if it cannot happen the `if` itself - not its body - is obsolete.

Reply:
:+1:

Please add error-handling:
* `If-Unmodified-Since`-Header missing -> Http-Status `428` (please also update openapi-spec accordingly)
* `If-Unmodified-Since`-Header too old (does not match) -> Http-Status `412`
* Header not parsable (not a valid date): Http-Status  `400`
* `Id` does not match a vocabulary: Http-Status `404`